### PR TITLE
send final state to listeners before game over

### PR DESCRIPTION
**Pull request by xerus2000**
Should fix https://github.com/CAU-Kiel-Tech-Inf/gui/issues/38

### CODE REVIEW

File: sdk/src/server-api/sc/framework/plugins/AbstractGame.kt [hunk 1]
```diff
@@ -130,17 +130,16 @@ abstract class AbstractGame<P : Player>(override val pluginUUID: String) : IGame
      * - sends out a state update and requests a new move
      */
     protected fun next(nextPlayer: P?) {
+        // if paused, notify observers only (so they can update the GUI appropriately)
+        notifyOnNewState(currentState, isPaused)
```
Comment (SKoschnicke) [inline sdk/src/server-api/sc/framework/plugins/AbstractGame.kt:134]: are these only listeners with admin privileges which are notified here or could my client register itself as an observer and receive the new state on a paused game earlier?

Comment (xerus2000) [reply sdk/src/server-api/sc/framework/plugins/AbstractGame.kt:134]: this didn't change anything regarding that, only that it is now also sent before game over

Comment (SKoschnicke) [reply sdk/src/server-api/sc/framework/plugins/AbstractGame.kt:134]: Yes, observation is only possible with admin rights, checked in sc/server/Lobby.kt:75

**Review by SKoschnicke — APPROVED** [review]

**Review by anarchuser — APPROVED** [review]
looks good, thanks

**Pull request merged**